tests/unit/test_state.py

```diff
@@ -775,6 +802,138 @@ def test_state_output_unifier_result_converted_to_false(self):
         assert statedecorators.OutputUnifier("unify")(lambda: data)()["result"] is False
 
 
+@skipIf(pytest is None, "PyTest is missing")
+class SubStateReturnsTestCase(TestCase):
+    """
+    TestCase for code handling state returns.
+    """
+
+    def test_sub_state_output_check_changes_is_dict(self):
+        """
+        Test that changes key contains a dictionary.
+        :return:
+        """
+        data = {"changes": {}, "sub_state_run": {"": {"changes": []}}}
+        out = statedecorators.OutputUnifier("content_check")(lambda: data)()
+        assert "'Changes' should be a dictionary" in out["sub_state_run"][""]["comment"]
+        assert not out["sub_state_run"][""]["result"]
+
+    def test_sub_state_output_check_return_is_dict(self):
+        """
+        Test for the entire return is a dictionary
+        :return:
+        """
+        data = {"sub_state_run": {"": ["whatever"]}}
+        out = statedecorators.OutputUnifier("content_check")(lambda: data)()
+        assert (
+            "Malformed state return. Data must be a dictionary type"
+            in out["sub_state_run"][""]["comment"]
+        )
+        assert not out["sub_state_run"][""]["result"]
+
+    def test_sub_state_output_check_return_has_nrc(self):
```

Comment:
Can we expand `nrc`? I guess it is expanded in the comment which might be good enough.